### PR TITLE
[SPARK-12318][SPARKR] Save mode in SparkR should be error by default

### PR DESCRIPTION
**Pull request by zjffdu**
@shivaram  Please help review.

### CODE REVIEW

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -1903,7 +1903,7 @@ setMethod("except",
 #' }
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: could you add the default to this doc line above:
```
 #' @param mode One of 'append', 'overwrite', 'error', 'ignore' save mode
```

Comment (zjffdu) [inline R/pkg/R/DataFrame.R]: @felixcheung  Is it necessary to do that ? Because I notice there's no doc for the default value in other R APIs. Actually user can see the default value in the function signature which is also in the R doc. 
```
## S4 method for signature 'DataFrame,character'
write.df(df, path, source = NULL,
  mode = "error", ...)
```

**Comment by sun-rui** [comment]
Document this change into migration guide?

**Comment by zjffdu** [comment]
Make sense, will do.

**Comment by felixcheung** [comment]
I don't think we have a migration guide section in SparkR programming guide but we should, like sql:
http://spark.apache.org/docs/latest/sql-programming-guide.html#migration-guide

File: R/pkg/R/DataFrame.R (hunk 1 above)
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: Sure. I think that's just good practice, but it's fine both ways.

**Comment by zjffdu** [comment]
@felixcheung yes, there's no migration doc for R. So I just put the doc in the data frame example. I can create migration section if necessary.

**Comment by shivaram** [comment]
Yeah lets create a migration section -- This also came up in another PR and I think is useful to have.

**Comment by 3ourroom** [comment]
NAVER - http://www.naver.com/
--------------------------------------------

3ourroom@naver.com 님께 보내신 메일 <Re: [spark] [SPARK-12318][SPARKR] Save mode in SparkR should be error by default (#10290)> 이 다음과 같은 이유로 전송 실패했습니다.

--------------------------------------------

받는 사람이 회원님의 메일을 수신차단 하였습니다. 


--------------------------------------------

File: docs/sparkr.md [hunk 2]
```diff
@@ -148,7 +148,7 @@ printSchema(people)
 </div>
 
 The data sources API can also be used to save out DataFrames into multiple file formats. For example we can save the DataFrame from the previous example
-to a Parquet file using `write.df`
+to a Parquet file using `write.df` (Before spark 1.7, mode's default value is 'append', we change it to 'error' to be consistent with scala api)
```
Comment (shivaram) [inline docs/sparkr.md]: Could you reword this to 
```
Until Spark 1.6, the default mode for writes was `append`. It was changed in Spark {{site.SPARK_VERSION_SHORT}} to `error` to match the Scala API
```

File: docs/sparkr.md [hunk 3]
```diff
@@ -387,3 +387,10 @@ The following functions are masked by the SparkR package:
 Since part of SparkR is modeled on the `dplyr` package, certain functions in SparkR share the same names with those in `dplyr`. Depending on the load order of the two packages, some functions from the package loaded first are masked by those in the package loaded after. In such case, prefix such calls with the package name, for instance, `SparkR::cume_dist(x)` or `dplyr::cume_dist(x)`.
   
 You can inspect the search path in R with [`search()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/search.html)
+
+
+# Migration Guide
+
+## Upgrading From SparkR 1.6 to 1.7
+
+ - Before Spark 1.7, the default save mode is `append` in api saveDF/write.df/saveAsTable, it is changed to `error` to be consistent with scala api.
```
Comment (shivaram) [inline docs/sparkr.md]: Same as above

**Comment by shivaram** [comment]
One minor thing is that I'm not sure its very good to hard code these version numbers here. For example we could have a RC3 and then this might be in Spark 1.6

cc @JoshRosen Are there known good practices on how to handle documentation changes of this form ?

**Comment by shivaram** [comment]
Cc @marmbrus it'll be good to have this PR in rc3. There are some unaddressed docs comments but I can fix them up while merging ?

**Comment by marmbrus** [comment]
Sorry, RC3 has already been cut.

**Comment by shivaram** [comment]
LGTM. Merging into master (and not `branch-1.6` as we probably don't want to change the API in a minor release like 1.6.1).

**Comment by marmbrus** [comment]
@shivaram actually, i'm going to have to cut RC3 again.  You get a second chance if you merge now!

**Comment by shivaram** [comment]
Oh sure cherry picking now

**Comment by shivaram** [comment]
Done ! Thanks @marmbrus for the heads up
